adapter/src/components/LoginModal.js

```diff
@@ -6,57 +6,133 @@ import {
     ModalActions,
     Button,
     InputField,
+    NoticeBox,
 } from '@dhis2/ui'
 import PropTypes from 'prop-types'
 import React, { useState } from 'react'
 import i18n from '../locales/index.js'
-import { post } from '../utils/api.js'
+import { get, post, postJSON } from '../utils/api.js'
+import { styles } from './styles/LoginModal.style.js'
 
 // Check if base URL is set statically as an env var (typical in production)
 const staticUrl = process.env.REACT_APP_DHIS2_BASE_URL
 
-export const LoginModal = ({ appName, baseUrl }) => {
+const getUseNewLoginAPI = async (server) => {
+    try {
+        // if loginConfig is available, the instance can use new endpoints
+        await get(`${server}/api/loginConfig`)
+        return true
+    } catch (e) {
+        // if loginConfig is not available, the instance must use old endpoints
+        return false
+    }
```

Comment:
I'm going to `console.error` the error in case it's something beside it just doesn't exist (or if we get an error back on an instance that should have it)